ide/coqide/coqOps.ml

```diff
@@ -361,6 +362,30 @@ object(self)
     let query = Coq.query (route_id,(phrase,sid)) in
     Coq.bind (Coq.seq action query) next
 
+  method proof_diff where ~next : unit Coq.task =
+    (* todo: would be nice to ignore comments, too *)
+    let rec back iter =
+      if iter#is_start then iter
+      else
+        let c = iter#char in
+        if Glib.Unichar.isspace c || c = 0 then back (iter#backward_char)
+        else if c = int_of_char '.' then iter#backward_char
+        else iter in
+
+    let where = back (buffer#get_iter_at_mark where) in
+    let until _ start stop =
+      (buffer#get_iter_at_mark stop)#compare where >= 0 &&
+      (buffer#get_iter_at_mark start)#compare where <= 0 in
+    let state_id = fst @@ self#find_id until in
+    if Stateid.to_int state_id = 1 then begin
+     next (Interface.Good (Pp.str "Put the cursor over proven lines for \"Show Proof\" differences"))
+    end else
+      let diff_opt = Interface.(match Coq.PrintOpt.(get diff) with
+        | StringValue diffs -> diffs
+        | _ -> "off") in
```

Comment:
I assume that you want the default to be, say `removed`, in order to be consistent with what is written in the reference manual.

Reply:
The default is "off".  What are you looking at in the reference manual?

Reply:
I misread the documentation above.

However, since this comment is an explicit request to show the proof with diff, why to have a dependency in the diff setting?

Alternatively, if this key prints diff only if diffs are on, it should simply be called `Show Proof`, not `Show Proof Diffs`, no?

Reply:
Yeah, I considered that.  But it makes it less obvious that you can get diffs, which users have to become aware of.  So I'll make the menu say `Show Proof (with Diffs, if set)`